### PR TITLE
Make things work without registering the dtype.

### PR DESCRIPTION
**Pull request by hameerabbasi**
@scopatz Unfortunately 2.9 and below have a bug -- We need H5Py 2.10 at least. But it works with 2.10 release.

### CODE REVIEW

File: bth5/dataset.py [hunk 1]
```diff
@@ -7,27 +7,102 @@
 import h5py
 import numpy as np
 import numba as nb
+import json
+
+
+class DatasetView(h5py.Dataset):
```
Comment (scopatz) [inline bth5/dataset.py]: Please add a docstring

**Comment by scopatz** [comment]
Why did the tests fail with your last commit?

**Comment by hameerabbasi** [comment]
Formatting. Re-ran black.

**Comment by scopatz** [comment]
got it

**Comment by scopatz** [comment]
Thanks @hameerabbasi!

**Pull request merged**